tomviz/DataSource.h

```diff
@@ -177,6 +177,14 @@ class DataSource : public QObject
   /// Set the active scalars by array name.
   void setActiveScalars(const QString& arrayName);
   QString activeScalars() const;
+  /// Set the active scalars by component index.
+  void setActiveScalars(int arrayIdx);
+  int activeScalarsIdx() const;
```

Comment:
I think we should consider being a little more explicit with the API, and call it Index rather than Idx. It doesn't save that many characters.